junit-jupiter-params/src/main/java/org/junit/jupiter/params/provider/CsvArgumentsProvider.java

```diff
@@ -47,9 +47,20 @@ public void accept(CsvSource annotation) {
 
 	@Override
 	public Stream<? extends Arguments> provideArguments(ExtensionContext context) {
+		Preconditions.condition(this.annotation.value().length > 0 ^ !this.annotation.textBlock().isEmpty(),
+			() -> "@CsvSource must be declared with either `value` or `textBlock` but not both");
+
+		String[] lines;
+		if (!this.annotation.textBlock().isEmpty()) {
+			lines = this.annotation.textBlock().split("\\n");
```

Comment:
We could use `String.lines()` if we're running on JDK 11 here. Not sure it's worth it, though.

Reply:
Yeah, I actually used `.lines()` first and then realized it wouldn't work on JDK 8. So I reverted that change.

I know we use reflection like that in `JRE.determineCurrentVersion()` to use `java.lang.Runtime.Version`, but I don't think this particular use case warrants the use of reflection just to use `String.lines()`.

We can, however, precompile the regular expression for a slight performance boost. I'll go ahead and do that.

If people complain that they notice a considerable performance hit when using `textBlock`, we can always implement an improved algorithm at a later date.

Reply:
> We can, however, precompile the regular expression for a slight performance boost. I'll go ahead and do that.

I just force pushed that change.